File: backend/api/schemas/workforce.py
"""
api/schemas/workforce.py
========================
Pydantic response schemas for the Workforce Analytics router.
These define exactly what the API sends to the frontend — nothing more.
"""

from pydantic import BaseModel


class HiringTrendPoint(BaseModel):
    """One data point on the yearly hiring trend chart."""
    year: int
    total_postings: int
    avg_salary_usd: float | None
    remote_postings: int


class CountryHiringStats(BaseModel):
    """Hiring demand aggregated per country."""
    country_name: str
    country_code: str
    region: str
    total_postings: int
    avg_salary_usd: float | None


class IndustryHiringStats(BaseModel):
    """Hiring demand aggregated per industry."""
    industry_name: str
    sector: str
    total_postings: int
    avg_salary_usd: float | None
    ai_adoption_index: float


class RemoteBreakdown(BaseModel):
    """Remote vs on-site split for a given year."""
    total_postings: int
    remote_postings: int
    onsite_postings: int
    remote_pct: float


class TopRole(BaseModel):
    """A single job role ranked by posting volume."""
    role_name: str
    role_category: str
    seniority_level: str
    total_postings: int
    avg_salary_usd: float | None
    is_remote_eligible: bool

File: frontend/src/types/workforce.ts
// ─────────────────────────────────────────
// types/workforce.ts
// Field names mirror backend/api/schemas/workforce.py exactly.
// ─────────────────────────────────────────

// GET /api/v1/workforce/hiring-trends
export interface HiringTrendPoint {
  year: number
  total_postings: number
  avg_salary_usd: number | null
  remote_postings: number
}

// GET /api/v1/workforce/by-country
export interface CountryHiringStats {
  country_name: string
  country_code: string
  region: string
  total_postings: number
  avg_salary_usd: number | null
}

// GET /api/v1/workforce/by-industry
export interface IndustryHiringStats {
  industry_name: string
  sector: string
  total_postings: number
  avg_salary_usd: number | null
  ai_adoption_index: number
}

// GET /api/v1/workforce/remote-stats
export interface RemoteBreakdown {
  total_postings: number
  remote_postings: number
  onsite_postings: number
  remote_pct: number
}

// GET /api/v1/workforce/top-roles
export interface TopRole {
  role_name: string
  role_category: string
  seniority_level: string
  total_postings: number
  avg_salary_usd: number | null
  is_remote_eligible: boolean
}

// Aggregate type for the page hook
export interface WorkforceData {
  hiringTrends: HiringTrendPoint[]
  byCountry: CountryHiringStats[]
  byIndustry: IndustryHiringStats[]
  remoteStats: RemoteBreakdown
  topRoles: TopRole[]
}